### ISSUE TITLE
Adding an intellij project to local git gives code analysis errors on commit

### ISSUE THREAD

**Issue by darrellpf** [flutter/flutter#9505]
## Steps to Reproduce

Create a new Intellij flutter project
VCS/Import into version control/Create GIT repository
Commit changes on unversioned files
Browse and add unversioned files
Select all the files
Do the commit
Code analysis dialog box says there are errors

![image](https://cloud.githubusercontent.com/assets/2779084/25253665/bbc573f8-25d6-11e7-964c-b08ccb6ab1bb.png)

If you review the errors things are very mysterious and seem like there is a big problem.
If you just commit the project works fine

## Flutter Doctor
[✓] Flutter (on Linux, channel master)
    • Flutter at /home/darrell/flutter
    • Framework revision ca2bf1efd0 (3 hours ago), 2017-04-20 11:37:10 -0700
    • Engine revision a5b64899c9
    • Tools Dart version 1.23.0-dev.11.7

[✓] Host Executable Compatibility
    • Downloaded executables execute on host

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.2)
    • Android SDK at /home/darrell/Android/Sdk
    • Platform android-25, build-tools 25.0.2
    • Java binary at: /home/darrell/android-studio/jre/bin/java
    • Java version: OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[✓] Android Studio (version 2.4)
    • Android Studio at /home/darrell/android-studio
    • Gradle version 3.4.1
    • Java version: OpenJDK Runtime Environment (build 1.8.0_112-release-b736)

[✓] IntelliJ IDEA Community Edition (version 2017.1)
    • Dart plugin version 171.4249.16
    • Flutter plugin version 12.1

[✓] Connected devices
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 7.1.1 (API 25) (emulator)

**Comment by mit-mit** [comment]
I can repro this. The errors and warnings are:

```
**/Users/mit/tmp/testapp/android/gradle.properties**
Warning:(1, 1) Unused property
**/Users/mit/tmp/testapp/android/app/src/main/AndroidManifest.xml**
Error:(1, 2) Cannot find the declaration of element 'manifest'.
Error:(1, 26) URI is not registered (Settings | Languages & Frameworks | Schemas and DTDs)
Error:(3, 5) Attribute android:versionCode is not allowed here
Error:(4, 5) Attribute android:versionName is not allowed here
Error:(19, 91) Attribute android:icon is not allowed here
Error:(21, 19) Attribute android:launchMode is not allowed here
Error:(22, 19) Attribute android:theme is not allowed here
Error:(23, 19) Attribute android:configChanges is not allowed here
Error:(24, 19) Attribute android:hardwareAccelerated is not allowed here
Error:(25, 19) Attribute android:windowSoftInputMode is not allowed here
Error:(32, 3) Cannot find the declaration of element 'manifest'.
**/Users/mit/tmp/testapp/android/settings.gradle**
Warning:(6, 27) Constructor call is ambiguous
**/Users/mit/tmp/testapp/android/app/src/main/java/com/yourcompany/testapp/MainActivity.java**
Error:(9, 3) Annotations are not supported at this language level
Error:(9, 4) Usage of API documented as @since 1.4+
**/Users/mit/tmp/testapp/android/app/build.gradle**
Warning:(5, 29) Method call is ambiguous
Warning:(11, 5) Cannot assign 'GradleException' to 'Throwable'
```

**Comment by mit-mit** [comment]
I cannot repro this anymore. I believe it may have been fixed as a side effect of https://github.com/flutter/flutter/pull/10928.

@darrellpf can you try to `flutter upgrade`, create a new project, and see if things are working as expected for you?

**Comment by darrellpf** [comment]
First, when the Android framework detected message comes up it is necessary to click the dialog and then accept the Android manifest xml file. If you don't there are quite a few errors from the xml file.

Even with that there is still a few warnings and one error

 
![image](https://user-images.githubusercontent.com/2779084/27807082-75324d7e-5ff3-11e7-9501-a2dcdb203901.png)

[✓] Flutter (on Linux, locale en_US.UTF-8, channel master)
    • Flutter at /home/darrell/flutter
    • Framework revision 5f20f83415 (28 hours ago), 2017-07-02 09:34:31 -0700
    • Engine revision 8ad1576f27
    • Tools Dart version 1.24.2

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /home/darrell/Android/Sdk
    • Platform android-26, build-tools 26.0.0
    • ANDROID_HOME = /home/darrell/Android/Sdk
    • Java binary at: /home/darrell/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] Android Studio (version 2.4)
    • Android Studio at /home/darrell/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1)
    • Flutter plugin version 15.0
    • Dart plugin version 171.4694.29

[✓] Connected devices
    • None

**Comment by mit-mit** [comment]
Did some digging. These errors are unrelated to source control; a smaller set of repro steps are:

1. Start IntelliJ IDEA
1. Create a new Flutter project
1. Configure Java support as per https://flutter.io/intellij-ide/#edit-android-code
1. Invoke the menu item 'Analyze > Inspect Code...'

Get a number of errors, warnings, and lint issues.

**Comment by Hixie** [comment]
I'm leaving this in the current milestone since it's sort of a tool new user blocker (at least for users who are going to write OEM-side code), and it's sort of an IDE getting started blocker, and it's sort of engine maintenance...

**Comment by mit-mit** [comment]
Yes, let's leave. I have a hunch this is caused by the same root issue as https://github.com/flutter/flutter-intellij/issues/1482
